lib/gateways/s3.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Gateway that provides documents that belong to dropboxes from Amazon S3.
+ */
+export default ({
+  client,
+  bucketName,
+  configuration: { urlPrefix, maxUploadBytes }
+}) => {
+  function getDownloadUrl(key) {
+    console.log('Generating pre-signed download url', { key });
+    return client.getSignedUrl('getObject', {
+      Bucket: bucketName,
+      Key: key,
+      Expires: 60
+    });
+  }
+
+  /**
+   * Fetches all documents that have been uploaded to a given dropbox.
+   * @param {String} dropboxId the id of the dropbox
+   */
+  function getByDropboxId(dropboxId) {
```

Comment:
I think we can remove this function

Reply:
Whether or not we need this is part of how we store the documents discussion

Reply:
Based on the previous discussion this can now be removed.